sechub-wrapper-secret-validator/src/test/java/com/mercedesbenz/sechub/wrapper/secret/validator/execution/SecretValidatorExecutionContextFactoryTest.java

```diff
@@ -0,0 +1,158 @@
+
+import com.mercedesbenz.sechub.wrapper.secret.validator.properties.SecretValidatorPDSJobResult;
+import com.mercedesbenz.sechub.wrapper.secret.validator.properties.SecretValidatorProperties;
+
+class SecretValidatorExecutionContextFactoryTest {
+
+    private SecretValidatorExecutionContextFactory factoryToTest;
+
+    @BeforeEach
+    void beforeEach() {
+        factoryToTest = new SecretValidatorExecutionContextFactory();
+    }
+
+    @Test
+    void not_existing_pds_job_result_file_throws_exception() {
+        /* prepare */
+        File notExisting = mock(File.class);
+        when(notExisting.exists()).thenReturn(false);
+        SecretValidatorPDSJobResult secretValidatorPDSJobResult = new SecretValidatorPDSJobResult();
+        secretValidatorPDSJobResult.setFile(notExisting);
+
+        factoryToTest.pdsResult = secretValidatorPDSJobResult;
+
+        /* execute + test */
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> factoryToTest.create());
+        assertEquals("PDS job result file: " + notExisting + " does not exist!", exception.getMessage());
+    }
+
+    @Test
+    void not_readable_pds_job_result_file_throws_exception() {
+        /* prepare */
+        File notReadable = mock(File.class);
+        when(notReadable.exists()).thenReturn(true);
+        when(notReadable.canRead()).thenReturn(false);
+        SecretValidatorPDSJobResult secretValidatorPDSJobResult = new SecretValidatorPDSJobResult();
+        secretValidatorPDSJobResult.setFile(notReadable);
+
+        factoryToTest.pdsResult = secretValidatorPDSJobResult;
+
+        /* execute + test */
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> factoryToTest.create());
+        assertEquals("PDS job result file: " + notReadable + " is not readable!", exception.getMessage());
+    }
+
+    @Test
+    void invalid_sarif_pds_job_result_file_throws_exception() {
+        /* prepare */
+        File invalidFile = new File("src/test/resources/config-test-files/invalid-files/invalid-sarif.txt");
+        SecretValidatorPDSJobResult secretValidatorPDSJobResult = new SecretValidatorPDSJobResult();
+        secretValidatorPDSJobResult.setFile(invalidFile);
+
+        factoryToTest.pdsResult = secretValidatorPDSJobResult;
+
+        /* execute + test */
+        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> factoryToTest.create());
+        assertEquals("Creating SARIF report model from: " + invalidFile + " failed!", exception.getMessage());
+    }
+
+    @Test
+    void not_existing_secret_validator_config_file_throws_exception() {
+        /* prepare */
+        // we need a valid Sarif report because the pds job result is read first
+        File validFile = new File("src/test/resources/config-test-files/valid-files/test-result.txt");
+        SecretValidatorPDSJobResult secretValidatorPDSJobResult = new SecretValidatorPDSJobResult();
```

Comment:
Just a hint: You could mock the SecretValidatorPDSJobResult - here you test another implementation as well...

Also the SecretValidatorPDSJobResult could be set inside a field in a `@BeforeEach` method  would reduce boilerplate code